src/hooks/useQueryParams.tsx

```diff
@@ -0,0 +1,27 @@
+import { useMemo } from 'react';
+import { useRouter } from 'next/navigation';
+
+
+type QueryParams<T extends Record<string, string>> = {
+  [K in keyof T]: (value: T[K]) => void;
+};
+
+export const useQueryParams = <T extends Record<string, string>>(params: T): QueryParams<T> => {
+  const router = useRouter();
+
+  return useMemo(() => {
```

Comment:
I'm new to using `useMemo`. As I understand it, this is caching the "page" so that it isn't recalculated when navigating back/forward I assume? Let me know if there was another purpose that I missed. Thanks for adding this.